Makefile

```diff
@@ -1,241 +1,582 @@
+	@echo ""
+	@echo "Available Targets by Category:"
+	@awk 'BEGIN {FS = ":.*##"; current_group = ""} \
+		/^##@/ { \
+			group = substr($$0, 5); \
+			gsub(/^[ \t]+|[ \t]+$$/, "", group); \
+			if (group != current_group) { \
+				current_group = group; \
+				printf "\n\033[1m%s\033[0m\n", group; \
+			} \
+		} \
+		/^[a-zA-Z_-]+:.*?##/ { \
+			target = $$1; \
+			desc = $$2; \
+			gsub(/^[ \t]+|[ \t]+$$/, "", desc); \
+			printf "  \033[36m%-28s\033[0m %s\n", target, desc; \
+		}' $(MAKEFILE_LIST)
+
+# ==================== Quick Start ====================
+##@ Quick Start
+
+.PHONY: setup
+setup: install-tools validate-env ## Initial project setup
+
+.PHONY: all
+all: clean build test ## Clean, build, and test everything
+
+# ==================== Primary Build Targets ====================
+##@ Build
+
+.PHONY: build
+build: build-api-server build-cli build-toolkit build-init ## Build all binaries
+
+.PHONY: build-api-server
+build-api-server: validate-env ## Build API server binary
+	@echo "Building API server ($(GOOS)/$(GOARCH))..."
+	@CGO_ENABLED=0 GOOS=$(GOOS) GOARCH=$(GOARCH) $(GO) build \
+		$(GOFLAGS) \
+		-ldflags='$(LD_FLAGS_API)' \
+		-o $(API_SERVER_BIN) \
+		cmd/api-server/main.go
+	@echo "API server built: $(API_SERVER_BIN)"
+
+.PHONY: build-cli
+build-cli: validate-env ## Build CLI binary (kubectl-testkube)
+	@echo "Building CLI ($(GOOS)/$(GOARCH))..."
+	@CGO_ENABLED=0 GOOS=$(GOOS) GOARCH=$(GOARCH) $(GO) build \
+		$(GOFLAGS) \
+		-ldflags='$(LD_FLAGS_API)' \
+		-o $(CLI_BIN) \
 		cmd/kubectl-testkube/main.go
-
-docker-build-api:
-	env SLACK_BOT_CLIENT_ID=** SLACK_BOT_CLIENT_SECRET=** ANALYTICS_TRACKING_ID=** ANALYTICS_API_KEY=** SEGMENTIO_KEY=** CLOUD_SEGMENTIO_KEY=** DOCKER_BUILDX_CACHE_FROM=type=registry,ref=docker.io/kubeshop/testkube-api-server:latest ALPINE_IMAGE=alpine:3.20.6 goreleaser release -f goreleaser_files/.goreleaser-docker-build-api.yml --rm-dist --snapshot
-
-docker-build-cli:
-	env SLACK_BOT_CLIENT_ID=** SLACK_BOT_CLIENT_SECRET=** ANALYTICS_TRACKING_ID=** ANALYTICS_API_KEY=** SEGMENTIO_KEY=** CLOUD_SEGMENTIO_KEY=** DOCKER_BUILDX_CACHE_FROM=type=registry,ref=docker.io/kubeshop/testkube-cli:latest ALPINE_IMAGE=alpine:3.20.6 goreleaser release -f .builds-linux.goreleaser.yml --rm-dist --snapshot
-
-#make docker-build-executor EXECUTOR=zap GITHUB_TOKEN=*** DOCKER_BUILDX_CACHE_FROM=type=registry,ref=docker.io/kubeshop/testkube-zap-executor:latest
-#add ALPINE_IMAGE=alpine:3.20.6 env var for building of curl and scraper executor
-docker-build-executor:
-	goreleaser release -f goreleaser_files/.goreleaser-docker-build-executor.yml --clean --snapshot
-
-dev-install-local-executors:
-	kubectl apply --namespace testkube -f https://raw.githubusercontent.com/kubeshop/testkube-operator/main/config/samples/executor_v1_executor.yaml
-
-install-swagger-codegen-mac:
-	brew install swagger-codegen
-
-openapi-generate-model: openapi-generate-model-testkube
-
-openapi-generate-model-testkube:
```

Comment:
why is it removed?

Reply:
I renamed it to `generate-openapi` - https://github.com/kubeshop/testkube/pull/6519/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R382

Works the same, manually tested it.